ruby/ext/google/protobuf_c/defs.c

```diff
@@ -374,6 +393,23 @@ static VALUE Descriptor_msgclass(VALUE _self) {
   return self->klass;
 }
 
+/*
+ * call-seq:
+ *     Descriptor.options => options
+ *
+ * Returns the `MessageOptions` for this `Descriptor`.
+ */
+static VALUE Descriptor_options(VALUE _self) {
+  Descriptor* self = ruby_to_Descriptor(_self);
+  const google_protobuf_MessageOptions* opts = upb_MessageDef_Options(self->msgdef);
+  upb_Arena* arena = upb_Arena_New();
+  size_t size;
+  char* serialized = google_protobuf_MessageOptions_serialize(opts, arena, &size);
+  VALUE message_options = decode_options("MessageOptions", size, serialized, self->descriptor_pool);
```

Comment:
We lost the caching. Could you use `rb_ivar_get()` and `rb_ivar_set()` to add back the caching we had before? This could go into `decode_options()` if you pass `self` as a parameter.